contracts/base/InterestConsensus.sol

```diff
@@ -18,134 +18,139 @@ pragma experimental ABIEncoderV2;
 
         require(
-            !nodeSubmissions[lender][blockNumber].finalized,
-            "INTEREST_ALREADY_FINALIZED"
+            !signerNonceTaken[response.signer][response.signature.signerNonce],
+            "SIGNER_NONCE_TAKEN"
         );
+        signerNonceTaken[response.signer][response.signature.signerNonce] = true;
 
-        bytes32 hashedData = _hashData(
-            lender,
-            blockNumber,
-            interest,
-            signature.signerNonce
-        );
-        require(_signatureValid(signature, hashedData), "SIGNATURE_NOT_VALID");
-
-
-            ZeroCollateralCommon.AggregatedInterest memory aggregatedData
-         = nodeSubmissions[lender][blockNumber];
-
-        // if this is the first submission for this request
-        if (aggregatedData.totalSubmissions == 0) {
-            aggregatedData = ZeroCollateralCommon.AggregatedInterest({
-                totalSubmissions: 1,
-                minValue: interest,
-                maxValue: interest,
-                sumOfValues: interest,
-                finalized: false
-            });
-        } else {
-            if (interest < aggregatedData.minValue) {
-                aggregatedData.minValue = interest;
-            }
-            if (interest > aggregatedData.maxValue) {
-                aggregatedData.maxValue = interest;
-            }
-
-            aggregatedData.sumOfValues = aggregatedData.sumOfValues.add(interest);
-            aggregatedData.totalSubmissions++;
-        }
+        require(response.responseTime >= now.sub(THIRTY_DAYS), "RESPONSE_EXPIRED");
 
-        emit InterestSubmitted(msg.sender, lender, blockNumber, interest);
+        bytes32 responseHash = _hashResponse(response, requestHash);
+        require(
+            _signatureValid(response.signature, responseHash, response.signer),
+            "SIGNATURE_INVALID"
+        );
 
-        if (aggregatedData.totalSubmissions >= requiredSubmissions) {
-            aggregatedData.finalized = true;
+        interestSubmissions[request.lender][request.endTime].addValue(response.interest);
 
-            // average the submissions
-            uint256 finalInterest = aggregatedData.sumOfValues.div(
-                aggregatedData.totalSubmissions
-            );
+        emit InterestSubmitted(
+            response.signer,
+            request.lender,
+            request.endTime,
+            response.interest
+        );
+    }
 
-            require(
-                _resultsWithinTolerance(
-                    aggregatedData.maxValue,
-                    aggregatedData.minValue,
-                    finalInterest
-                ),
-                "MAXIMUM_TOLERANCE_SURPASSED"
+    function _hashResponse(
+        ZeroCollateralCommon.InterestResponse memory response,
+        bytes32 requestHash
+    ) internal pure returns (bytes32) {
+        return
+            keccak256(
+                abi.encode(
+                    response.responseTime,
```

Comment:
The reason I only put `lenders` is because the Lenders contract has a fixed `interestConsensus` address. So by specifying the lenders address, you are implicitly stating which interestConsensus address you are sending to

Reply:
The hash for `hashResponse` includes the hash of the request. So by putting `lenders` in the request, it is also inside the response hash too.